### PR TITLE
Update sdba documentation

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [ ] Tests for the changes have been added (for bug fixes / features)
  - [x] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Merge of the documentation of the module (`xclim/sdba/__init__.py`) with the rst (`docs/sdba.rst`). The former was not appearing in any easy-to-find place.
* Some cleanup, addition of the Roy et al. citation to the bib file.
* Addition of a "Other documentation" section that points to the notebooks, but also to some github issues discussing some aspects of SDBA.

### Does this PR introduce a breaking change?
No

### CODE REVIEW

File: docs/sdba.rst [hunk 1]
```diff
@@ -2,22 +2,64 @@
 Bias adjustment and downscaling algorithms
 ==========================================
 
-`xarray` data structures allow for relatively straightforward implementations of simple bias-adjustment and downscaling algorithms documented in :ref:`Adjustment Methods <sdba:SDBA User API>`.
-Each algorithm is split into `train` and `adjust` components. The `train` function will compare two DataArrays `x` and `y`, and create a dataset storing the *transfer* information allowing to go from `x` to `y`.
-This dataset, stored in the adjustment object, can then be used by the `adjust` method to apply this information to `x`. `x` could be the same `DataArray` used for training, or another `DataArray` with similar characteristics.
 
-For example, given a daily time series of observations `ref`, a model simulation over the observational period `hist` and a model simulation over a future period `sim`, we would apply a bias-adjustment method such as *detrended quantile mapping* (DQM) as::
+The `xclim.sdba` submodule provides bias-adjustment methods and will eventually provide statistical downscaling algorithms.
```
Comment (huard) [inline docs/sdba.rst:6]:
```suggestion
The `xclim.sdba` submodule provides a collection of bias-adjustment methods meant to meant to correct for systematic biases found in climate model simulations relative to observations. 
```

File: docs/sdba.rst [hunk 2]
```diff
@@ -2,22 +2,64 @@
...
+Almost all adjustment algorithms conform to the `train` - `adjust` scheme, formalized within `TrainAdjust` classes.
```
Comment (huard) [inline docs/sdba.rst:7]:
```suggestion
Almost all adjustment algorithms conform to the `train` - `adjust` scheme, meaning that adjustment factors are first estimated on training data sets, then applied in a distinct step to the data to be adjusted.  
```

File: docs/sdba.rst [hunk 3]
```diff
@@ -2,22 +2,64 @@
...
-  from xclim import sdba
+  # Create the adjustment object by training it with reference and model data, plus certain arguments
+  Adj = Adjustment.train(ref, hist, group="time.month")
+  # Get a scenario by applying the adjustment to a simulated timeseries.
+  scen = Adj.adjust(sim, interp="linear")
+  Adj.ds.af  # adjustment factors.
 
-  dqm = sdba.adjustment.DetrendedQuantileMapping.train(ref, hist)
-  scen = dqm.adjust(sim)
+Most method can either be applied additively by multiplication.
```
Comment (huard) [inline docs/sdba.rst:18]:
```suggestion
Most method support both additive and multiplicative correction factors.
```

File: docs/sdba.rst [hunk 4]
```diff
@@ -2,22 +2,64 @@
...
+Also, the `group` argument allows adjustment factors to be estimated independently for different periods: the full
+time series,  months, seasons or day of the year. The `interp` argument then allows for interpolation between these
```
Comment (huard) [inline docs/sdba.rst:20]:
```suggestion
time series,  months, seasons or day of the year.  For monthly groupings, the `interp` argument then allows for interpolation between
```

File: docs/sdba.rst [hunk 5]
```diff
@@ -2,22 +2,64 @@
...
+adjustment factors to avoid discontinuities in the bias-adjusted series (only applicable for monthly grouping).
```
Comment (huard) [inline docs/sdba.rst:21]:
```suggestion
adjustment factors to avoid discontinuities in the bias-adjusted series.
```

File: docs/sdba.rst [hunk 6]
```diff
@@ -2,22 +2,64 @@
...
+- creating and training the adjustment object ``Adj = Adjustment.train(obs, sim, **kwargs)`` (from :py:mod:`xclim.sdba.adjustment`)
+- adjustment ``scen = Adj.adjust(sim, **kwargs)``
+- post-processing on ``scen`` (for example: re-trending)
+
+The train-adjust approach allows to inspect the trained adjustment object. The training information is stored in
+the underlying `Adj.ds` dataset and usually has a `af` variable with the adjustment factors. Its layout and the
+other available variables vary between the different algorithm, refer to :ref:`Adjustment methods <sdbauserapi>`.
+
+Parameters needed by the training and the adjustment are saved to the ``Adj.ds`` dataset as a `adj_params` attribute.
+Other parameters, those only needed by the adjustment are passed in the `adjust` call and written to the history
```
Comment (huard) [inline docs/sdba.rst:44]: unclear

File: docs/sdba.rst [hunk 7]
```diff
@@ -2,22 +2,64 @@
...
+
+.. _grouping:
+
+Grouping
+========
+
+For basic time period grouping (months, day of year, season), passing a string to the methods needing it is sufficient.
+Most methods acting on grouped data also accept a `window` int argument to pad the groups with data from adjacent ones.
+Units of `window` are the sampling frequency of the main grouping dimension (usually `time`). For more complex grouping,
+one can pass an instance of :py:class:`xclim.sdba.base.Grouper` directly.
```
Comment (huard) [inline docs/sdba.rst:55]: I think it would help to see and example here of a grouping declaration.

File: docs/sdba.rst [hunk 8]
```diff
@@ -41,6 +83,44 @@ Radiation and precipitation
 
 In theory, short wave radiation should be capped when precipitation is not zero, but there is as of yet no mechanism proposed to do that, see :cite:t:`hoffmann_meteorologically_2012`.
 
+Other documentation
```
Comment (huard) [inline docs/sdba.rst:86]:
```suggestion
Usage examples
```

File: docs/sdba.rst [hunk 9]
```diff
@@ -41,6 +83,44 @@ Radiation and precipitation
...
+===================
+The usage of this module is documented in two example notebooks: `SDBA <notebooks/sdba.ipynb>`_ and `SDBA advanced <notebooks/sdba-advanced.ipynb>`_.
+
+Some issues were also discussed on the Github repository. Most of these are still open questions, feel free to participate to the discussion!
```
Comment (huard) [inline docs/sdba.rst:90]: Discussion topics
==============

**Review by huard — APPROVED** [review]

File: docs/sdba.rst [hunk 10]
```diff
@@ -83,7 +83,7 @@ Radiation and precipitation
 
 In theory, short wave radiation should be capped when precipitation is not zero, but there is as of yet no mechanism proposed to do that, see :cite:t:`hoffmann_meteorologically_2012`.
 
-Other documentation
+Usage examples
 ===================
```
Comment (Zeitsperre) [inline docs/sdba.rst]:
```suggestion
==============
```

**Pull request merged**